tools/reviewer/remote_server/RemoteServer.java

```diff
@@ -0,0 +1,206 @@
+      return parseGoogleResponse(response);
+    }
+
+    @Override
+    public void handle(HttpExchange httpExchange) throws IOException {
+      if ("post".equalsIgnoreCase(httpExchange.getRequestMethod())) {
+        logger.atInfo().log("Handling initial auth request");
+
+        InitialAuthRequest.Builder request = InitialAuthRequest.newBuilder();
+        JsonFormat.parser().merge(getPostParamsString(httpExchange), request);
+        AuthResponse response = null;
+
+        try {
+          response = requestTokensFromGoogle(request.build());
+        } catch (Exception e) {
+          e.printStackTrace();
+          httpExchange.sendResponseHeaders(HTTP_STATUS_CODE_UNAUTHORIZED, -1);
+        }
+
+        sendProto(httpExchange, response);
+      }
+    }
+  }
+
+  static class RefreshTokenHandler implements HttpHandler {
+    private String clientId;
+    private String clientSecret;
+
+    RefreshTokenHandler(String clientId, String clientSecret) {
+      this.clientId = clientId;
+      this.clientSecret = clientSecret;
+    }
+
+    private String buildRefreshParams(String refreshToken) {
+      return String.format(
+          "client_id=%s&client_secret=%s&refresh_token=%s&grant_type=refresh_token",
+          this.clientId, this.clientSecret, refreshToken);
+    }
+
+    private AuthResponse tokenRefreshRequest(RefreshTokenRequest request) throws Exception {
+      URL url = new URL(RemoteServer.TOKEN_URL);
+      String response = httpPost(url, buildRefreshParams(request.getRefreshToken()));
+      return parseGoogleResponse(response);
+    }
+
+    @Override
+    public void handle(HttpExchange httpExchange) throws IOException {
+      if ("post".equalsIgnoreCase(httpExchange.getRequestMethod())) {
+        logger.atInfo().log("Handling refresh token request");
+
+        RefreshTokenRequest.Builder req = RefreshTokenRequest.newBuilder();
+        JsonFormat.parser().merge(getPostParamsString(httpExchange), req);
+        AuthResponse response = null;
+        try {
+          response = tokenRefreshRequest(req.build());
+        } catch (Exception e) {
+          e.printStackTrace();
+          httpExchange.sendResponseHeaders(HTTP_STATUS_CODE_UNAUTHORIZED, -1);
+        }
+
+        sendProto(httpExchange, response);
+      }
+    }
+  }
+
+  private RemoteServer(String clientId, String clientSecret) throws IOException {
+    httpServer = HttpServer.create(new InetSocketAddress(port.get()), 0);
+    httpServer.createContext("/gcode", new InitialAuthHandler(clientId, clientSecret));
+    httpServer.createContext("/refresh", new RefreshTokenHandler(clientId, clientSecret));
+  }
+
+  public void start() {
+    this.httpServer.start();
+  }
+
+  public static void main(String[] args) throws IOException {
+    Flags.parseCurrentPackage(args);
+
+    RemoteServer server =
+        new RemoteServer(
+            System.getenv("REMOTESERVER_CLIENT_ID"), System.getenv("REMOTESERVER_CLIENT_SECRET"));
```

Comment:
Let's use a flag for these. We try to avoid env vars since they rely on system state and thus are less reproducible.